components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -324,20 +326,17 @@ protected AuthenticationContext initializeFlow(HttpServletRequest request,
 
         List<String> result = Collections.emptyList();
         String requestType = request.getParameter(FrameworkConstants.RequestParams.TYPE);
-        if (StringUtils.isNotBlank(request.getParameter("acr_values")) && !"null"
-                .equals(request.getParameter("acr_values"))) {
-            String[] acrValues = request.getParameter("acr_values").split(" ");
-            AuthenticationMethodNameTranslator translator = FrameworkServiceDataHolder
-                    .getInstance().getAuthenticationMethodNameTranslator();
+        List<String> acrValuesList = (List<String>) request.getAttribute(ACR_VALUES_ATTRIBUTE);
+        if (CollectionUtils.isNotEmpty(acrValuesList)) {
+            AuthenticationMethodNameTranslator translator = FrameworkServiceDataHolder.getInstance()
```

Comment:
We can get rid of this AuthenticationMethodNameTranslator, and associated logic around it. This was a remnant of the previous "Conditional Authentication" work, which is not based on Javascript.

Reply:
This is done by [PR 1430](https://github.com/wso2/carbon-identity-framework/pull/1430)